### PR TITLE
Setting max/default ActiveDeadlineSeconds on the deployer pod

### PR DESCRIPTION
**Pull request by abhgupta**

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
What controls how long a LifecycleHook can run?  My understanding is that that can also cause a new pod to be created.

**Comment by abhgupta** [comment]
@derekwaynecarr Good catch! I am now setting the same value on the deployment hook pod as well. These would apply to the deployment strategies that we provide. Users can still go ahead and provide their own deployer pod image and we'd have no control over anything they do in there. But for the deployment strategies that we provide, we have added the attribute.

File: pkg/deploy/strategy/support/lifecycle.go [hunk 1]
```diff
@@ -143,7 +146,8 @@ func buildContainer(hook *deployapi.ExecNewPodHook, deployment *kapi.Replication
 					Resources:  resources,
 				},
 			},
-			RestartPolicy: kapi.RestartPolicyNever,
+			ActiveDeadlineSeconds: &maxDeploymentDurationSeconds,
```
Comment (derekwaynecarr) [inline pkg/deploy/strategy/support/lifecycle.go]: test that verifies this?

File: pkg/deploy/controller/deployment/controller.go [hunk 2]
```diff
@@ -169,7 +172,8 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 					Resources: deploymentConfig.Template.Strategy.Resources,
 				},
 			},
-			RestartPolicy: kapi.RestartPolicyNever,
+			ActiveDeadlineSeconds: &maxDeploymentDurationSeconds,
```
Comment (derekwaynecarr) [inline pkg/deploy/controller/deployment/controller.go]: is there a test in controller_test.go that can verify this is set?

**Comment by abhgupta** [comment]
Also, I'd like to keep these commits separate.

File: pkg/deploy/strategy/support/lifecycle.go (hunk 1 above)
Comment (abhgupta) [inline pkg/deploy/strategy/support/lifecycle.go]: Added

File: pkg/deploy/controller/deployment/controller.go (hunk 2 above)
Comment (abhgupta) [inline pkg/deploy/controller/deployment/controller.go]: Added

**Comment by abhgupta** [comment]
@derekwaynecarr Please review/merge

**Comment by derekwaynecarr** [comment]
[merge]

**Pull request merged**